### PR TITLE
Small formatting changes.

### PR DESCRIPTION
**Pull request by NinjaLikesCheez**
This brings the repo more in line with python standards and fixes a small issue by creating the log directory if it doesn't already exist. By upgrading to PySide6, Apple Silicon devices are supported too :D 

Things to test:

* Does piOS support PySide6? 
* Does the UI remain the same with PySide6 vs PySide2?

### CODE REVIEW

**Comment by fieldOfView** [comment]
Thanks for the PR. "Small formatting changes" he says... I would hate to see your "big formatting changes" ;-)

I am fairly certain Raspberry Pi OS still does not support PySide6, unfortunately. So we are going to have to stay compatible with PySide2 (since that RPi by the door is the main target). We could _also_ add PySide2 support, by way of trying to import PySide6 and catching the ImportError.

I must say I would have preferred two separate PRs; one for the formatting changes, and one for the PySide6 upgrade. In the same vein I'm a fan of more atomic commits too.

**Comment by NinjaLikesCheez** [comment]
Heh, I'll pull out the PySide changes so we just have the formatting changes + log directory creation. While 'large' for a single commit, the formatting changes are rather tightly related - but for the future, I'll make sure to have more atomic commits.

**Comment by NinjaLikesCheez** [comment]
I've made the change and force pushed over the top to modify the commit to have only those changes :)

**Comment by fieldOfView** [comment]
I have mostly learned Python by contributing to one project (Ultimaker Cura), and they have a code-style that is not 100% PEP8 aligned. I am certainly open to improvements (and could use a slap on the wrist if I do things "suboptimally", so I appreciate the code style suggestions. This whole project is a learning exercise for me.

A large part of the changes is renaming the python source files. I am used to a "one class per file" policy and naming the file the same as the class. But it looks like PEP8 does indeed describe lowercase names for modules: https://peps.python.org/pep-0008/#package-and-module-names . There's also this though:

> Underscores can be used in the module name if it improves readability (...) , although the use of underscores is discouraged.

For me, `pixeldash` would still be readable (and preferred over `pixel_dash`), but I understand the readability benefit of the `*_sensor` modules. I would keep something like `vcgencmd_sensor` (since `vcgencmd` is the actual command that is being called), and would even prefer `spacestate` over `space_state`.  But I guess that brings us into matters of taste.   

In this PR, you are replacing f-strings with %-formatting. I thought %-formatting is now discouraged. Any reason you prefer %-formatting? I do appreciate the changes where I'm adding strings together (here's where that slap on the wrist would come in).

File: requirements.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
 PySide2
 requests
 psutil
-RPI.GPIO
\ No newline at end of file
+RPI.GPIO; sys_platform == 'linux'
```
Comment (fieldOfView) [inline requirements.txt]: This is a good enhancement, but not all linuxes are raspberry pi os. According to https://peps.python.org/pep-0508/#environment-markers we should also be able to limit the `platform_machine` - to `armv7l` - which would further narrow it down.

File: src/sensors/Sensor.py [hunk 2]
```diff
@@ -42,24 +42,24 @@ def _processResponse(self) -> None:
 
     def _updateData(self) -> None:
         self._timestamp = time.monotonic()
-        self.dataChanged.emit()
+        self.data_changed.emit()
 
-    dataChanged = Signal()
+    data_changed = Signal()
```
Comment (fieldOfView) [inline src/sensors/Sensor.py]: I am a bit torn here. If the Signal `dataChanged` is changed to `data_changed`, then the Property `unitMap` should also be `unit_map`. But for me, both of them map to things properties/"things" in QML, and QML expects its own code style. For example the eventhandler for a signal with the name `data_changed` would become `onData_changed` which looks weird (whereas the eventhandler for `dataChanged` becomes `onDataChanged`).

File: requirements.txt (hunk 1 above)
Comment (NinjaLikesCheez) [reply requirements.txt]: Good call, I'll dig in and see if we can grab some additional values to check as (iirc) there's also an ARM64 version of piOS.

File: src/sensors/Sensor.py (hunk 2 above)
Comment (NinjaLikesCheez) [reply src/sensors/Sensor.py]: That's entirely reasonable, if it lines up with the QML then I will change those back and disable linting for them instead.

**Comment by NinjaLikesCheez** [comment]
> I have mostly learned Python by contributing to one project (Ultimaker Cura), and they have a code-style that is not 100% PEP8 aligned. I am certainly open to improvements (and could use a slap on the wrist if I do things "suboptimally", so I appreciate the code style suggestions. This whole project is a learning exercise for me.

In a similar vein, these are entirely suggestions and there's 100% no onus on you to agree or accept any of them. It's been a couple years since I've written 'production-level' Python so of my changes may be outdated!

> A large part of the changes is renaming the python source files. I am used to a "one class per file" policy and naming the file the same as the class. But it looks like PEP8 does indeed describe lowercase names for modules: https://peps.python.org/pep-0008/#package-and-module-names . There's also this though:
> 
> > Underscores can be used in the module name if it improves readability (...) , although the use of underscores is discouraged.
> 
> For me, `pixeldash` would still be readable (and preferred over `pixel_dash`), but I understand the readability benefit of the `*_sensor` modules. I would keep something like `vcgencmd_sensor` (since `vcgencmd` is the actual command that is being called), and would even prefer `spacestate` over `space_state`. But I guess that brings us into matters of taste.

Now that you mention it, I would agree - grouping the related terms together when the name is PixelDash and not Pixel Dash. I will make this change too.

> In this PR, you are replacing f-strings with %-formatting. I thought %-formatting is now discouraged. Any reason you prefer %-formatting? I do appreciate the changes where I'm adding strings together (here's where that slap on the wrist would come in).

It's an optimization (albeit an incredibly small and insignificant one, especially in the context of this project). Using f-string interpolation in a logging call means that python has to interpolate the string prior to passing it through to the function where as using `%s` delays interpolation until the logging function decides it is necersarry to do so. 

For example, if you're logging something at a `DEBUG` level - but the project level is set to `INFO` - the message would be interpolated regardless of whether the logging framework will actually log the message.

**Comment by NinjaLikesCheez** [comment]
fwiw, I will likely make these changes on Wednesday unless I find some time before then :D 

Changes to make:

- Find mapped QML element names in the python side and change them back from snake casing to camel casing
- Investigate restricting the `RPI.GPIO` to _just_ piOS
- Undo & fixup the naming changes around `pixeldash`, `statespace`, commands, sensors, and emitters to group together the 'name' portion

File: __main__.py [hunk 3]
```diff
@@ -1,28 +1,36 @@
 import sys
 import os
```
Comment (fieldOfView) [inline __main__.py:2]: This import is now unused, but I can fix that later.
Edit: Nevermind, it wasn't

**Review by fieldOfView — APPROVED** [review]

**Pull request merged**